lib/mixlib/install/backend/artifactory.rb

```diff
@@ -19,72 +19,141 @@
+            uri << result["repo"]
+            uri << result["path"]
+            uri << result["name"]
+            { "downloadUri" => uri.join("/") }.merge(map_properties(result["properties"]))
+          end
         end
 
-        def info
-          begin
-            results = client.get("/api/search/prop", params, headers)["results"]
-          rescue Errno::ETIMEDOUT => e
-            raise e, "unstable channel uses endpoint #{ARTIFACTORY_ENDPOINT} \
-which is currently only accessible through Chef's internal network."
+        #
+        # Returns all version found for channel and product_name
+        #
+        def product_versions
+          query = <<-QUERY
+items.find(
+  {"repo": "#{repository}"},
+  {"@omnibus.project": "#{options.product_name}"}
+).include("name", "path", "repo", "property")
+          QUERY
+
+          results = aql_post(query)
+          properties = results["results"].collect { |result| result["properties"] }
+          versions = properties.collect do |prop|
+            prop.first["value"] if prop.first["key"] == "omnibus.version"
           end
+          versions.compact
+        end
 
-          if options.platform
-            artifact(results.first)
+        #
+        # Parse latest version
+        #
+        def product_version
+          if options.latest_version?
+            Mixlib::Versioning.find_target_version(product_versions, nil, false, true).to_s
           else
-            results.collect do |result|
-              artifact(result)
-            end
+            options.product_version
           end
         end
 
         private
 
-        def artifact(result)
+        def artifact(artifact_map)
           ArtifactInfo.new(
-            md5:              result["properties"]["omnibus.md5"].first,
-            sha256:           result["properties"]["omnibus.sha256"].first,
-            version:          result["properties"]["omnibus.version"].first,
-            platform:         result["properties"]["omnibus.platform"].first,
-            platform_version: result["properties"]["omnibus.platform_version"].first,
-            architecture:     result["properties"]["omnibus.architecture"].first,
-            url:              result["downloadUri"]
+            md5:              artifact_map["omnibus.md5"],
+            sha256:           artifact_map["omnibus.sha256"],
+            version:          artifact_map["omnibus.version"],
+            platform:         artifact_map["omnibus.platform"],
+            platform_version: artifact_map["omnibus.platform_version"],
+            architecture:     artifact_map["omnibus.architecture"],
+            url:              artifact_map["downloadUri"]
           )
         end
 
-        def params
-          params = {
-            "repos" => ARTIFACTORY_REPOSITORY,
-            "omnibus.version" => options.product_version
-          }
+        #
+        # Creates a client based on the required ENVs:
+        #   ARTIFACTORY_ENDPOINT
```

Comment:
I think we **must** have some working defaults for these. @schisamo maybe we can have a read-only account that we can hardcode here so that not every user of mixlib-install is required to set these. If we can't do that I am not sure if requiring every user of mixlib-install to set these is viable.